python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_collection.py

```diff
@@ -0,0 +1,212 @@
+            *[container_proxy.delete_item(item=get_key(key), partition_key=get_partition_key(key)) for key in keys],
+            return_exceptions=True,
+        )
+        exceptions = [result for result in results if isinstance(result, Exception)]
+        if exceptions:
+            raise MemoryConnectorException("Failed to delete item(s).", exceptions)
+
+    @override
+    def _serialize_dicts_to_store_models(self, records: Sequence[dict[str, Any]], **kwargs: Any) -> Sequence[Any]:
+        serialized_records = []
+
+        key_field_name = self.data_model_definition.key_field_name
+        for record in records:
+            serialized_record = {**record, COSMOS_ITEM_ID_PROPERTY_NAME: record[key_field_name]}
+            if key_field_name != COSMOS_ITEM_ID_PROPERTY_NAME:
+                # Remove the key field from the serialized record
+                serialized_record.pop(key_field_name, None)
+
+            serialized_records.append(serialized_record)
+
+        return serialized_records
+
+    @override
+    def _deserialize_store_models_to_dicts(self, records: Sequence[Any], **kwargs: Any) -> Sequence[dict[str, Any]]:
+        deserialized_records = []
+
+        key_field_name = self.data_model_definition.key_field_name
+        for record in records:
+            if COSMOS_ITEM_ID_PROPERTY_NAME not in record:
+                raise VectorStoreModelDeserializationException(
+                    f"The record does not have the {COSMOS_ITEM_ID_PROPERTY_NAME} property."
+                )
+
+            deserialized_record = {**record, key_field_name: record[COSMOS_ITEM_ID_PROPERTY_NAME]}
+            if key_field_name != COSMOS_ITEM_ID_PROPERTY_NAME:
+                # Remove the id property from the deserialized record
+                deserialized_record.pop(COSMOS_ITEM_ID_PROPERTY_NAME, None)
+
+            deserialized_records.append(deserialized_record)
+
+        return deserialized_records
+
+    @override
+    async def create_collection(self, **kwargs) -> None:
+        try:
+            database_proxy = await self._get_database_proxy()
+            await database_proxy.create_container_if_not_exists(
+                id=self.collection_name,
+                partition_key=self.partition_key,
+                indexing_policy=kwargs.pop(
+                    "indexing_policy", create_default_indexing_policy(self.data_model_definition)
+                ),
+                vector_embedding_policy=kwargs.pop(
+                    "vector_embedding_policy", create_default_vector_embedding_policy(self.data_model_definition)
+                ),
+                **kwargs,
+            )
+        except Exception as e:
+            raise MemoryConnectorException("Failed to create container.") from e
+
+    @override
+    async def does_collection_exist(self, **kwargs) -> bool:
+        try:
+            database = await self._get_database_proxy()
+            containers = database.query_containers(
```

Comment:
from database_proxy.create_container_if_not_exists:
```suggestion
            try:
                container_proxy = self.get_container_client(id)
                await container_proxy.read(
                    session_token=session_token,
                    initial_headers=initial_headers,
                    **kwargs
                )
                return True
            except CosmosResourceNotFoundError:
                return False
```